first-party/jib-native-image-extension-maven/README.md

```diff
@@ -0,0 +1,57 @@
+# Jib GraalVM Native Image Extension
+
+This extension containerizes a [GraalVM native-imgae](https://www.graalvm.org/docs/reference-manual/native-image/) application configured with [Native Image Maven Plugin](https://www.graalvm.org/docs/reference-manual/native-image/#integration-with-maven).
+
+The extension expects the `native-image-maven-plugin` to do the the heavy lifting of generating a "native image" (with the `native-image:native-image` goal). (Here, "image" refers to an executable binary, not a container image.) Then the extension simply copies the binary, say, `<project root>/target/com.example.mymainclass`, into a container image and sets executable bits. It also auto-sets the container image entrypoint to the binary, say, `/app/com.example.mymainclass` (unless you manually configure `<container><entrypoint>` in the main Jib configuration).
```

Comment:
`<project root>/target/com.example.mymainclass` seems like a strange example for a binary? Is that normal graalvm native image naming?

Reply:
Yeah, if you don't configure the binary name in `native-image-maven-plugin`, it uses a fully qualified Java main class in lowercase for an executable name.